src/tag.cpp

```diff
@@ -89,62 +89,6 @@ HSTag* find_unused_tag() {
     return nullptr;
 }
 
-int tag_remove_command(int argc, char** argv, Output output) {
-    // usage: remove TAG [TARGET]
-    // it removes an TAG and moves all its wins to TARGET
-    // if no TARGET is given, current tag is used
-    if (argc < 2) {
-        return HERBST_NEED_MORE_ARGS;
-    }
-    HSTag* tag = find_tag(argv[1]);
-    HSTag* target = (argc >= 3) ? find_tag(argv[2]) : get_current_monitor()->tag;
-    if (!tag) {
-        output << argv[0] << ": Tag \"" << argv[1] << "\" not found\n";
-        return HERBST_INVALID_ARGUMENT;
-    } else if (!target) {
-        output << argv[0] << ": Tag \"" << argv[2] << "\" not found\n";
-    } else if (tag == target) {
-        output << argv[0] << ": Cannot merge tag \"" << argv[1] << "\" into itself\n";
-        return HERBST_INVALID_ARGUMENT;
-    }
-    if (find_monitor_with_tag(tag)) {
-        output << argv[0] << ": Cannot merge the currently viewed tag\n";
-        return HERBST_TAG_IN_USE;
-    }
-    // prevent dangling tag_previous pointers
-    all_monitors_replace_previous_tag(tag, target);
-    // save all these windows
-    vector<HSClient*> buf;
-    tag->frame->foreachClient([&buf](HSClient* client) {
-        buf.push_back(client);
-    });
-    for (auto client : buf) {
-        stack_remove_slice(client->tag()->stack, client->slice);
-        client->setTag(target);
-        stack_insert_slice(client->tag()->stack, client->slice);
-        ewmh_window_update_tag(client->window_, client->tag());
-        target->frame->insertClient(client);
-    }
-    tag->frame = shared_ptr<HSFrame>();
```

Comment:
I've dropped this step. At first it just seemed pointless to me now that I'm `delete`ing the tag, but this also caused a segfault in `applyLayout()` when (a couple of calls down) the frame got accessed once more.

Reply:
I meant to say that I didn't carry the assignment of `tag->frame` to the new implementation because it seemed superfluous to me, given the (now) explicit deletion of the tag object.

Besides that, I discovered that the existing code even segfaulted *due to* this assignment. So maybe this was a bug anyway, but that's not what motivated me to drop the assignment.

That said, I'm not 100% sure whether there was some importance to that assignment that I might have missed, and that's why I wanted to draw your attention to it.

Reply:
I see, thanks! I guess the double assignment simply came from an inconsistent / unfinished migration to the `HSTag` class.